stdlib/sql/to.go

```diff
@@ -216,38 +230,109 @@ func (t *ToSQLTransformation) Finish(id execute.DatasetID, err error) {
+
+func correctBatchSize(batchSize, numberCols int) int {
+	/*
+		BatchSize for the DB is the number of parameters that can be queued within each call to Exec.
+		As each row you send has a parameter count equal to the number of columns (i.e. the number of "?" used in the insert statement), and some DBs
+		have a default limit on the number of parameters which can be queued before calling Exec; SQLite, for example, has a default of 999 (can only be changed
+		at compile time).
+
+		So if the row width is 10 columns, the maximum Batchsize would be:
+
+		(999 - row_width) / row_width = 98 rows. (with 0.9 of a row unused)
+
+		and,
+
+		(1000 - row_width) / row_width = 99 rows. (no remainder)
+
+		NOTE: Given a statement like:
+
+		INSERT INTO data_table (now,values,difference) VALUES(?,?,?)
+
+		each iteration of EXEC() would add 3 new values (one for each of the '?' placeholders) - but the final "parameter count" includes the initial 3 column names.
+		this is why the calculation subracts an initial "column width" from the supplied Batchsize.
+
+		Sending more would result in the call to Exec returning a "too many SQL variables" error, and the transaction would be rolled-back / aborted
+	*/
+
+	if batchSize < numberCols {
+		// if this is because the width of a single row is very large, pass to DB driver, and if this exceeds the number of allowed parameters
+		// this will be fed back to the user to handle - possibly by reducing the row width
+		return numberCols
+	}
+	return (batchSize - numberCols) / numberCols
+}
+
+func getTranslationFunc(driverName string) (func() translationFunc, error) {
+	// simply return the translationFunc that corresponds to the driver type
+	switch driverName {
+	case "sqlite3":
+		return SqliteColumnTranslateFunc, nil
+	case "postgres", "sqlmock":
+		return PostgresColumnTranslateFunc, nil
+	case "mysql":
+		return MysqlColumnTranslateFunc, nil
+	default:
+		return nil, errors.Newf(codes.Internal, "invalid driverName: %s", driverName)
+	}
+
+}
+
 func CreateInsertComponents(t *ToSQLTransformation, tbl flux.Table) (colNames []string, valStringArray [][]string, valArgsArray [][]interface{}, err error) {
 	cols := tbl.Cols()
+	batchSize := correctBatchSize(t.spec.Spec.BatchSize, len(cols))
+
 	labels := make(map[string]idxType, len(cols))
 	var questionMarks, newSQLTableCols []string
 	for i, col := range cols {
 		labels[col.Label] = idxType{Idx: i, Type: col.Type}
 		questionMarks = append(questionMarks, "?")
 		colNames = append(colNames, col.Label)
-
+		driverName := t.spec.Spec.DriverName
+		// the following allows driver-specific type errors (of which there can be MANY) to be returned, rather than the default of invalid type
+		translateColumn, err := getTranslationFunc(driverName)
+		if err != nil {
+			return nil, nil, nil, err
+		}
 		switch col.Type {
 		case flux.TFloat:
-			newSQLTableCols = append(newSQLTableCols, fmt.Sprintf("%s FLOAT", col.Label))
+			v, err := translateColumn()(col.Type, col.Label)
```

Comment:
Maybe we can use `fallthrough` here since the `translateColumn()` code for all the types is the same?

Reply:
done - no logic changes at all, just collapsed switch into shorter version.